nodes/geometry/bmesh_tesselate.py

```diff
@@ -0,0 +1,43 @@
+import bpy
+from bpy.props import *
+from ... events import propertyChanged
+from ... base_types.node import AnimationNode
+
+class BMeshTessellateNode(bpy.types.Node, AnimationNode):
+    bl_idname = "an_BMeshTessellateNode"
+    bl_label = "Triangulate BMesh"
+    bl_width_default = 160
+    searchLabels = ["Tesselate BMesh"]
+
+    quad = IntProperty(name = "Quad Method", update = propertyChanged, max = 3, min = 0)
+    ngon = IntProperty(name = "Ngon Method", update = propertyChanged, max = 1, min = 0)
+
+    def create(self):
+        self.newInput("BMesh", "BMesh", "bm")
+        self.newOutput("BMesh", "BMesh", "bm")
+        self.newOutput("Integer List", "Matching Ngon Indices", "ngonIndices", hide = True)
+
+    def draw(self, layout):
+        layout.prop(self, "quad")
+        layout.prop(self, "ngon")
+
+    def getExecutionCode(self):
+        isLinked = self.getLinkedOutputsDict()
+        if not any(isLinked.values()): return
+        
+        yield "faces = bm.faces[:]"
+        
+        # do not invert isLinked order, this needs original faces
+        if isLinked["ngonIndices"]: 
+            yield "ngonIndices = list( range(len(faces)) )"
```

Comment:
just a little thing: these exra spaces aren't necessary in my opinion. Sometimes it is helpful to make code clearer this way, but I think this case here is not special enouph to break the (unwritten) rules.